### ISSUE TITLE
Persistent sessions do not seem to work

### ISSUE THREAD

**Issue by haphut** [k00mi/mqtt-hs#4]
Here's my example code:

``` hs
{-# LANGUAGE OverloadedStrings #-}

module Main where

import Control.Concurrent (threadDelay)
import qualified Data.ByteString as B
import Network.MQTT
import Network.MQTT.Logger

mqttConfig :: MQTTConfig
mqttConfig = defaultConfig
                 { cHost = "213.138.147.225"
                 , cClean = False
                 }

handleMessage :: Topic -> B.ByteString -> IO ()
handleMessage t bs = do
    print t
    print bs

main :: IO ()
main = do
    mqtt <- connect mqttConfig
    qos <- subscribe mqtt Handshake "/hfp/journey/#" handleMessage
    putStrLn $ "Got this QoS: " ++ show qos
    threadDelay $ round $ 5 * 1e6
```

If I compile it using Stackage 5.0 and run it, I get the following output:

``` text
[Debug] Sending CONNECT
[Debug] Received CONNACK
[Debug] Sending SUBSCRIBE
[Error] {handle: <socket: 3>}: recvLoop: end of file
[Error] recvLoop: No reconnect, terminating.
haskell-mqtt-test: thread blocked indefinitely in an MVar operation
```

Am I doing something wrong?

**Comment by k00mi** [comment]
Hi haphut,

it turns out it was the parsing of CONNACK messages that was incorrect.
The broker you use for testing rejects clients that try to establish
a persistent session, so this bug was only triggered when the `clean`
flag was `False`.

I fixed this in 1d6631e8a2265132f5501667db2df7fa98bedd67.
Can you make sure it works now?

**Comment by haphut** [comment]
I tested https://github.com/k00mi/mqtt-hs/commit/8e9947bd6fc89ed2ab5ff846c9258fc1c40711eb and it works great. Thank you for the explanation and the fix!
